examples/migration/create_complete_campaign_both_apis_phase_4.py

```diff
@@ -0,0 +1,349 @@
+        ad_group_operation.status = client.get_type('AdGroupAdStatusEnum',
+                                  version='v2').PAUSED
+        ad_group_operation.ad.expanded_text_ad.headline_part1.value = \
+                                    'Cruise to Mars #{}'.format(
+                                    str(uuid.uuid4())[:4])
+        ad_group_operation.ad.expanded_text_ad.headline_part2.value = \
+                                    'Best Space Cruise Line'
+        ad_group_operation.ad.expanded_text_ad.description.value = \
+                                    'Buy your tickets now!'
+        final_urls =  client.get_type('StringValue', version='v2')
+        final_urls.value = 'http://www.example.com'
+        ad_group_operation.ad.final_urls.extend([final_urls])
+        operations.append(operation)
+
+    adgroup_service = client.get_service('AdGroupAdService', version='v2')
+    ad_group_ad_response = adgroup_service.mutate_ad_group_ads(customer_id,
+                                                            operations)
+    new_ad_resource_names = []
+    for i in range(NUMBER_OF_ADS):
+        new_ad_resource_names.append(
+            ad_group_ad_response.results[i].resource_name)
+
+    new_ads = get_ads(client, customer_id, new_ad_resource_names)
+    for i in range(len(new_ads)):
+        print('Created expanded text ad with ID {}, status {} and '
+              'headline {}.{}'.format(new_ads[i].ad.id.value,
+              new_ads[i].status,
+              new_ads[i].ad.expanded_text_ad.headline_part1.value,
+              new_ads[i].ad.expanded_text_ad.headline_part2.value))
+
+
+def get_ads(client, customer_id, new_ad_resource_names):
+    """Retrieves a google.ads.google_ads.v2.types.AdGroupAd instance  .
+
+    Args:
+        client: A google.ads.google_ads.client.GoogleAdsClient instance.
+        customer_id: (str) Customer ID associated with the account.
+        new_ad_resource_names: (str) Resource name associated with the Ad group.
+
+    Returns:
+        An instance of the google.ads.google_ads.v2.types.AdGroupAd message
+            class of the newly created ad group ad.
+    """
+    def formatter(myst):
+        results = []
+        for i in myst:
+            results.append(repr(i))
+        return ','.join(results)
+    resouce_names = formatter(new_ad_resource_names)
+
+    ga_service = client.get_service('GoogleAdsService', version='v2')
+    query = ('SELECT ad_group_ad.ad.id, '
+             'ad_group_ad.ad.expanded_text_ad.headline_part1, '
+             'ad_group_ad.ad.expanded_text_ad.headline_part2, '
+             'ad_group_ad.status, ad_group_ad.ad.final_urls, '
+             'ad_group_ad.resource_name '
+             'FROM ad_group_ad '
+             'WHERE ad_group_ad.resource_name in ({}) '.
+             format(resouce_names))
+
+    response = ga_service.search(customer_id, query, PAGE_SIZE)
+    ad_group = list(response)
+    ads = []
+    for i in range(len(ad_group)):
```

Comment:
This could be more Pythonic by iterating over each row in response instead of casting to a list.